junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java

```diff
@@ -98,57 +103,53 @@ private String createSource(String[] lines) {
 		return String.join(System.lineSeparator(), lines);
 	}
 
-	private Worker getScriptExecutionWorker(ExtensionContext context) {
-		ExtensionContext root = context.getRoot();
-		ExtensionContext.Store rootStore = root.getStore(NAMESPACE);
-		return rootStore.getOrComputeIfAbsent(WORKER, this::createWorker, Worker.class);
-	}
-
-	// Create worker via reflection to hide the `javax.script` dependency.
-	private Worker createWorker(String name) {
-		logger.debug(() -> String.format("Creating instance of `%s`...", name));
-		Optional<Throwable> optionalThrowable = assumeScriptEngineIsLoadableByClassForName();
-		if (optionalThrowable.isPresent()) {
-			return new ThrowingWorker(optionalThrowable.get());
-		}
-		try {
-			return (Worker) Class.forName(name).getDeclaredConstructor().newInstance();
-		}
-		catch (ReflectiveOperationException cause) {
-			logger.error(cause, () -> String.format("Creating instance of `%s` failed", name));
-			return new ThrowingWorker(cause);
-		}
-	}
-
-	// This method may fail on JREs that don't provide the "javax.script" package at all
-	// or it fails on JREs launched with an active module system using insufficient module
-	// graphs, i.e. the application does not read "java.scripting" module.
-	Optional<Throwable> assumeScriptEngineIsLoadableByClassForName() {
-		try {
-			Class.forName("javax.script.ScriptEngine");
-			return Optional.empty();
-		}
-		catch (Throwable cause) {
-			logger.error(cause, () -> "Loading `javax.script.ScriptEngine` failed");
-			return Optional.of(cause);
-		}
-	}
-
 	/**
 	 * Evaluates scripts and returns a conditional evaluation result.
 	 */
-	interface Worker {
-		ConditionEvaluationResult work(ExtensionContext context, List<Script> scripts);
+	interface Evaluator {
+
+		ConditionEvaluationResult evaluate(ExtensionContext context, List<Script> scripts);
+
+		/**
+		 * Create evaluator via reflection to hide the `javax.script` dependency.
+		 */
+		static Evaluator forName(String name) {
+			Optional<Throwable> optionalThrowable = assumeScriptEngineIsLoadableByClassForName();
+			if (optionalThrowable.isPresent()) {
+				return new ThrowingEvaluator(optionalThrowable.get());
+			}
+			try {
+				return (Evaluator) Class.forName(name).getDeclaredConstructor().newInstance();
```

Comment:
I think it might be better to use `getClass().getClassLoader().loadClass(name)...`.

Reply:
Using the following new _private_ method for both cases:

		static Class<?> load(String name) throws ClassNotFoundException {
			return Evaluator.class.getClassLoader().loadClass(name);
		}

Reply:
Or revert to the initial code as it says in [Class.forName(String)](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#forName-java.lang.String-):
```
Returns the Class object associated with the class or interface with the given string name. Invoking this method is equivalent to:
  Class.forName(className, true, currentLoader)
where currentLoader denotes the defining class loader of the current class.
```

Reply:
sure 😛

Reply:
Or switch to `org.junit.platform.commons.util.ReflectionUtils.loadClass(String)` for consistency. 😇

Reply:
You pick!